tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -796,15 +793,21 @@ namespace Tensile
         }
 
         args.template append<uint32_t>("internalArgs", internalArg0);
+
+        int32_t internalArg1 = 0;
+        if(internalArgsSupport.wgm)
+        {
+            if(wgm == -1)
+                wgm = 1;
+            args.template append<int32_t>("internalArgs1", wgm);
```

Comment:
I am not sure if we really need -1024<= WGM <= 1024.
It might be OK to limit the range of WGM from -128 to 127 to fit 8bit range.
we can use S_SEXT_I32_I8 to convert signed int8 to 32bit.

Reply:
But, I think it is a good idea to expand the range of GSU to 16bit.
GSU>256 can be needed.
I think GSU range change is important and better to mention in your commit comment.

Reply:
I think we can keep this because we don't have any bits left.